src/Microsoft.Diagnostics.EventFlow.Core/Implementations/EventData.cs

```diff
@@ -28,6 +29,8 @@ public class EventData: IDeepCloneable<EventData>
 
         public long Keywords { get; set; }
 
+        public object EventContext { get; set; } = ContextResolver.GetRequestContext<object>();
```

Comment:
Depending on the EventFlow input implementation, there might be no context when the EventData is being created, or the context might be different from the context that was valid when original event was raised. 

A safer design would be to have the input fill the context. The implication is that EventData constructor probably does not need to set the context